gammapy/maps/core.py

```diff
@@ -887,8 +887,16 @@ def interp_to_geom(self, geom, preserve_counts=False, fill_value=0, **kwargs):
             data = map_copy.get_by_coord(coords)
             data = np.nan_to_num(data, nan=fill_value).astype(bool)
         else:
-            data = map_copy.interp_by_coord(coords, **kwargs)
-
+            data = map_copy.interp_by_coord(coords, fill_value=None, **kwargs)
+            if fill_value is not None: 
```

Comment:
This does not seem to be the right place for this fix to me. Because `.interp_by_coord()` and `.interp_by_pix()` will remain unaffected, even if the both define a `fill_value`. I think the better place is likely https://github.com/gammapy/gammapy/blob/master/gammapy/maps/wcs/ndmap.py#L139